src/access-decision-manager-express/access-decision-manager-provider.ts

```diff
@@ -0,0 +1,33 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { NextFunction, Request, Response } from 'express';
+import { AccessDecisionManager, Voter } from '../access-decision-manager';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type ADMArgumentFactory = (req: Request) => any;
+
+const AccessDecisionManagerProvider = (
+  getUser: ADMArgumentFactory,
```

Comment:
Is this order correct?
Or maybe would make more sense  this

voters, getUser, createContext

Why?
It could be the case when someone wants to use this module without specifying a user, so it would looked weird to call this with an empty fn

```(() => null, voters, ...) ```

But if we rearrange the params  would be something like

```(voters, ...)```